### ISSUE TITLE
Reduce clicks to print a label

### ISSUE THREAD

**Issue by mbucc** [thingybase/community#14]
It takes seven clicks to print a label.  Click on:
1. item
2. Label button
3. Jumbo option
4. Print button
5. brower File menu
6. Print option in browser file menu
7. Print button in print dialog

This could be reduced to three clicks:

1. save a default label style (e.g., "jumbo")
2. add a Print Label button that
    a. generates the PDF
    b. calls `window.print()`.
3. click Print button in print dialog

My work pattern was to pack a number of boxes, then print label for the day's worth of boxes packed.  (I use my office laserjet.)  Having to click so many times got pretty annoying.

**Comment by bradgessler** [comment]
Agreed! I'd like to get this to be as few clicks as possible. As far as the recommendation go, I ran into some issues with `window.print()` with PDFs when I tried this a while back. I'll give it another go and see where that's at.

Have you tried printing batch labels yet? It currently only works for "regular size" labels, but would this reduce clicks for your workflow if you're creating a bunch of items during the day, then printing the labels at night?

https://user-images.githubusercontent.com/4628/183520430-5d7dedfd-87b6-460c-9f65-231777b63db2.mov

It's my fault if you didn't see that, which is why I'm asking (I never blame users).

Some follow-up questions that can help me better understand your workflow:

* What kind of printer are you using? Exact make and model is the most helpful
* How many boxes do you pack in a day? How many labels do you then print at the end of the day?
* What kind of paper are you using to print the labels? Avery label sheets with adhesives? or just plain paper that you tape onto the box?
* What type of device and operating system are you using to create the items? Are you using that to also create the labels or do you use a different machine for that?

Answers to those help me better understand what you're trying to do so I can better streamline the workflow.

Thanks so much!

**Comment by bradgessler** [comment]
I did some research and determined that I should be able to eliminate the step that shows the PDF files before printing from the browser.

**Comment by mbucc** [comment]
Hi @bradgessler,

Sorry for the delayed response ... we were moving!  :)

> Have you tried printing batch labels yet?

No.  I would have used that if I knew about it, not sure why I missed it.

> What kind of printer are you using? Exact make and model is the most helpful

Brother MFC-7360N

> How many boxes do you pack in a day?

* packed at least one box on 27 different days
* median = 4 boxes/day
* min = 1, max = 26

> How many labels do you then print at the end of the day?

This varied.  In the beginning, I printed when a couple boxes were closed.  Towards the end, we stopped printing labels---just writing the box number was enough.  Sometimes I would stop packing the print labels before I started work the next day.

> What kind of paper are you using to print the labels? 
> Avery label sheets with adhesives? 
> or just plain paper that you tape onto the box?

The latter.

> What type of device and operating system are you using to create the items?

iPhones.  The dictation mode was really helpful for the items going in each box.

> Are you using that to also create the labels or do you use a different machine for that?

Yeah, printed from a macbook.   I've never figured out how to print from our phone and didn't bother for this task.

**Comment by bradgessler** [comment]
I did some research on this and determined `window.print()` won't work reliably as a solution to this problem. I'll have to consider other approaches or build a native application that can improve this print workflow.
